File: server/Identity.domain/Domain/RegisterModel.cs
using System.ComponentModel.DataAnnotations;

namespace Identity.Domain.Domain
{
    public class RegisterModel
    {
        [Required(ErrorMessage = "O email é obrigatório.")]
        [EmailAddress(ErrorMessage = "Email inválido.")]
        public string Email { get; set; }

        [Required(ErrorMessage = "A senha é obrigatória.")]
        [StringLength(100, MinimumLength = 8, ErrorMessage = "A senha deve ter no mínimo 8 caracteres.")]
        [RegularExpression(@"^(?=.*[A-Z])(?=.*[a-z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$",
            ErrorMessage = "A senha deve conter pelo menos 8 caracteres, incluindo uma letra maiúscula, uma letra minúscula, um número e um símbolo.")]
        public string Password { get; set; }

        [Required(ErrorMessage = "A confirmação de senha é obrigatória.")]
        [Compare("Password", ErrorMessage = "As senhas não coincidem.")]
        public string ConfirmPassword { get; set; }

        [Phone(ErrorMessage = "Número de telefone inválido.")]
        [Required(ErrorMessage = "O telefone é obrigatório.")]
        public string Telefone { get; set; }

        [Required(ErrorMessage = "A data de nascimento é obrigatória.")]
        [DataType(DataType.Date, ErrorMessage = "Data inválida.")]
        public DateTime? DataNascimento { get; set; }

        [StringLength(100, MinimumLength = 8, ErrorMessage = "O nome deve ter no mínimo 8 caracteres.")]
        [Required(ErrorMessage = "O telefone é obrigatório.")]
        public string Nome { get; set; }
    }
}

File: client/Identity.WebAPI/src/app/modules/account/model/RegisterModel.ts
import { FormGroup, FormControl, Validators } from '@angular/forms';

export class RegisterModel {
  email: string;
  password: string;
  confirmPassword: string;
  telefone: string;
  dataNascimento: Date;
  nome: string;

  constructor() {
    this.email = '';
    this.password = '';
    this.confirmPassword = '';
    this.telefone = '';
    this.dataNascimento = new Date();
    this.nome = '';
  }

  static getFormGroup(): FormGroup {
    return new FormGroup({
      email: new FormControl('', [
        Validators.required,
        Validators.email
      ]),
      password: new FormControl('', [
        Validators.required,
        Validators.minLength(8),
        Validators.pattern(/^(?=.*[A-Z])(?=.*[a-z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/)
      ]),
      confirmPassword: new FormControl('', [
        Validators.required,
        Validators.pattern(/^(?=.*[A-Z])(?=.*[a-z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/)
      ]),
      telefone: new FormControl('', [
        Validators.required,
        Validators.pattern(/^\(?\d{2}\)?\s?(\d{4,5})\s?-?\s?\d{4}$/)
      ]),
      dataNascimento: new FormControl('', [
        Validators.required,
        Validators.pattern(/\d{4}-\d{2}-\d{2}/)
      ]),
      nome: new FormControl('', [
        Validators.required,
        Validators.minLength(8)
      ])
    });
  }
}
